lib/web_ui/lib/src/engine/text/layout_service.dart

```diff
@@ -542,8 +661,53 @@ class LineBuilder {
         LineBreakResult.sameIndex(breakingPoint, LineBreakType.prohibited));
   }
 
+  LineBreakResult get _boxStart {
+    if (_boxes.isEmpty) {
+      return start;
+    }
+    // The end of the last box is the start of the new box.
+    return _boxes.last.end;
+  }
+
+  double get _boxLeft {
+    if (_boxes.isEmpty) {
+      return 0.0;
+    }
+    return _boxes.last.right;
+  }
+
+  ui.TextDirection get direction =>
+      paragraph.paragraphStyle._effectiveTextDirection;
+
+  /// Cuts a new box in the line.
+  ///
+  /// If this is the first box in the line, it'll start at the beginning of the
+  /// line. Else, it'll start at the end of the last box.
+  ///
+  /// A box should be cut whenever the end of line is reached, or when switching
+  /// from one span to another.
+  void cutBox() {
```

Comment:
placeBox , layoutBox, createBoxForRange?  cutBox sounds like it is splitting?

Reply:
Done.